### ISSUE TITLE
Feature Request: Add support for Music/"Artist - Album" folder structure in addition to Music/Artist/Album

### ISSUE THREAD

**Issue by anthonyrom** [lidarr/Lidarr#249]
### Description of request and what problem are you looking to solve?

As the title says. My library, and I'm sure many other peoples', is structured as Music/"Artist - Album". Lidarr appears to only support Music/Artist/Album, which makes it fail to import anything in my library. Seeing as (I believe) this is a popular way people structure their music collections, I think it would be beneficial for many to add support for this structure.

### Other Information

If this is already possible/supported, please let me know how it is done so I can see how much of a dumbass I am. :)

**Comment by bobloadmire** [comment]
out of curiosity why? this would be a lot hard to browse imo

**Comment by anthonyrom** [comment]
You're definitely right, especially with large libraries. However this is the recommended structure for Plex for untagged files/non-premium libraries, and I suspect that Plex users and Lidarr users have a good amount of overlap. See here: https://support.plex.tv/articles/200265296-adding-music-media-from-folders/

**Comment by bobloadmire** [comment]
Weird I'm using Plex with artist/album/song structure and have never had an
issue.

**Comment by Qstick** [comment]
Plex states this is for cases when there are no tags embedded in the files. I would venture that 95% of users have ID3 or other tags of some sort in their track files. I'm really not seeing any huge benefit to this, its going to be very low priority if we do it.

**Comment by jasii** [comment]
I prefer Artist - Album too. Specifically "Artist - Album Year - Album - Quality"
It's actually easier to search for an artist if I'm just browsing my directory and I am able to see all my albums  in alphabetical order and then even in release order.

+1 for this feature request

**Comment by Qstick** [comment]
Closing and tagging for long term look, this would require us to basically change everything in core and redefine how Lidarr works given we are dependent on Artist Folders.

**Comment by JohanSF** [comment]
I also very much prefer Artist - Album. A solution to the problem would be a feature in Lidarr with a checkbox like this:

[checked/unchecked] - Hardlink downloaded&imported album-folder to destination: [specify destination]

This way would be able to have a library in the format I like and it would be hardlinked to the same files on disk as the Lidarr-friendly library. I think this is the most elegant solution possible at the moment. Perhaps one day we can also get native support for different library schemes in Lidarr.

The above example could of course be made more general if many users find a need to do something with their albums after download&import.

**Comment by blackwind** [comment]
+1. I'm unable to import my library due to this poor design decision, but will try to use Lidarr in the meantime to dump future albums in my downloads folder so at least I don't have to keep tabs on, find, and download them myself.

**Comment by Qstick** [comment]
Guys,  feel free to submit a PR to add this functionality.

**Comment by annomatik** [comment]
+1. Just tried to add multiple artists to the same genre folder and failed. Please add this as an option for people who prefer the genre/artists - album structure. Why not give people a choice of what they prefer?

Thanks!

**Comment by annomatik** [comment]
Also, the only thing stopping people from being able to already do this, seems to be a simple check.

I have
- changed the the artist folder to a genre folder containing multiple existing albums from various artists
- told Lidarr to move the files, which it did
- told Lidarr to rescan the directory, which it also did

and as a result, Lidarr found all of the existing albums / tracks of that particular artist, without having any problem with the other existing directories / albums in that directory.

I tried to change a second artist to that genre folder and failed. I think the only thing stopping people from people using that feature is a simple check, if the artist folder has been already used for a different artist. Without that check, everyone could use a folder structure of his / her liking, without Lidarr breaking. (at least that's my impression)

**Comment by annomatik** [comment]
TL:DR I can currently add exactly one artist to the "final" genre folder, without breaking any of my existing stuff. To add the remaining artists and automate everything, Lidarr needs to be more flexible regarding multiple use of artist folders.

**Comment by Qstick** [comment]
Yes, however it's on hold for some importer changes to go in first.

**Comment by cbaken** [comment]
I do have the same setup, hope to see this supported soon.

**Comment by adam-ah** [comment]
Why would you close this @Qstick? This is a totally valid request

**Comment by anthonyrom** [comment]
For what it's worth, Plex has changed their recommendation to match the Artist/Album structure. Also, some time ago after this request basically got denied, I was able to use a tool to automatically restructure my library. I can't remember if I used Beets or MusicBrainz Picard, but it was relatively easy. I only had to manually reorganize the albums that it couldn't identify, which wasn't too bad. Still would be nice to have this feature for flexibility and users who don't want to go through that hassle to use Lidarr.
